webhook_to_fedora_messaging_messages/github/github.py

```diff
@@ -0,0 +1,96 @@
+# SPDX-FileCopyrightText: 2024 Contributors to the Fedora Project
+#
+# SPDX-License-Identifier: LGPL-3.0-or-later
+
+from ..base import Webhook2FedMsgBase
+from .utils import summarize_repository_event
+
+
+class GithubMessageV1(Webhook2FedMsgBase):
+    @property
+    def app_name(self):
+        return "Github"
+    
+    @property
+    def target_id(self):
+        return self.body['headers']['X-Github-Hook-Installation-Target-Id']
+    
+    @property
+    def signature(self):
+        """
+            SHA1 signature of the request.
+        """
+        return self.body['headers']['X-Hub-Signature']
+    
+    @property
+    def delivery(self):
+        """
+            A globally unique identifier (GUID) to identify the event.
+        """
+        return self.body['headers']['X-Github-Delivery']
+    
+    @property
+    def signature_sha256(self):
+        """
+            SHA1-256 signature of the request.
+        """
+        return self.body['headers']['X-Hub-Signature-256']
+    
+    @property
+    def agent_name(self):
+        """The username of the user who initiated the action that generated this message."""
+        return self.body['body']['sender']['login']
```

Comment:
It looks like the `['body']` key should not be there.

Reply:
I suggest reviewing this after inspecting the tests and etc.

Reply:
Correct, the body does have a `body` key, that confused me a bit :-)

That said, in Fedora Messaging messages, the agent name has to be a Fedora account username. So, it can't come from the Github webhook.
I think we should get the property from an `agent_name` key at the top level of the body and expect the main application to set it (after having converted the github username to a Fedora username).

Reply:
Should it be like in the base.py? Maybe it shouldn't be overridden in subclassses.